rosbag2_transport/test/rosbag2_transport/test_play_loop.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+// Copyright 2020, TNG Technology Consulting GmbH.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+
+#include <chrono>
+#include <future>
+#include <memory>
+#include <string>
+#include <thread>
+#include <vector>
+#include <utility>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "rosbag2_test_common/subscription_manager.hpp"
+
+#include "rosbag2_transport/rosbag2_transport.hpp"
+
+#include "test_msgs/msg/basic_types.hpp"
+#include "test_msgs/message_fixtures.hpp"
+
+#include "rosbag2_play_test_fixture.hpp"
+
+TEST_F(RosBag2PlayTestFixture, messages_played_in_loop) {
+  auto primitive_message1 = get_messages_basic_types()[0];
+  const int test_value = 42;
+  primitive_message1->int32_value = test_value;
+
+  auto topic_types = std::vector<rosbag2_storage::TopicMetadata>{
+    {"loop_test_topic", "test_msgs/BasicTypes", "", ""}
+  };
+
+  size_t num_messages = 3;
+  std::vector<std::shared_ptr<rosbag2_storage::SerializedBagMessage>> messages(num_messages,
+    serialize_test_message("loop_test_topic", 700, primitive_message1));
```

Comment:
what is 700 - looks like another magic value.

can you declare and define all your test constants at the top of the test so that their meaning and choice of values is obvious?